gc/base/MarkingScheme.hpp

```diff
@@ -169,43 +161,49 @@ class MM_MarkingScheme : public MM_BaseVirtual
 
 	void completeScan(MM_EnvironmentBase *env);
 	
-	MMINLINE void
-	scanObject(MM_EnvironmentBase *env, omrobjectptr_t objectPtr, MM_CollectorLanguageInterface::MarkingSchemeScanReason reason)
+	/**
+	 * Public. Called from external context, eg concurrent GC.
+	 */
+	MMINLINE uintptr_t
+	scanObject(MM_EnvironmentBase *env, omrobjectptr_t objectPtr, MM_MarkingSchemeScanReason reason, uintptr_t sizeToDo = UDATA_MAX)
 	{
-		if(MM_CollectorLanguageInterface::SCAN_REASON_PACKET == reason) {
-			env->_markStats._objectsScanned += 1;
+		GC_ObjectScannerState objectScannerState;
+		GC_ObjectScanner *objectScanner = _delegate.getObjectScanner(env, objectPtr, &objectScannerState, reason, &sizeToDo);
```

Comment:
I understand why the sizeToDo is passed so that it can be limited.  I feel like that complicates the scanner code and it will make it harder for implementers to implement their own scanner.